toffy/streak_detection.py

```diff
@@ -0,0 +1,409 @@
+                \n streak_mask\
+                \n streak_df\
+                \n filtered_streak_mask\
+                \n filtered_streak_df\
+                \n boxed_streaks\
+                \n corrected_streak_mask"
+            )
+        elif name is not None:
+            self._save(name)
+        else:
+            raise ValueError("Please enter a field name, or set `all_data` to True.")
+
+    def _get_save_dir(self, data_dir: Path, name: str, ext: str):
+        return Path(data_dir, name + f".{ext}")
+
+    def _save(self, name: str):
+        """Helper function for saving tiff binary masks and dataframes.
+
+        Args:
+            name (str): The field of the DataClass to be saved.
+        """
+        data_dir = Path(self.corrected_dir, f"streak_data_{self.streak_channel}")
+        if not os.path.exists(data_dir):
+            os.makedirs(data_dir)
+
+        data = getattr(self, name)
+        st = partial(self._get_save_dir, data_dir, name)
+
+        if type(data) is np.ndarray:
+            io.imsave(st("tiff"), data.astype(np.int8), check_contrast=False)
+        elif type(data) is pd.DataFrame:
+            data.to_csv(st("csv"))
+
+
+def _make_binary_mask(
+    input_image: np.ndarray,
+    gaussian_sigma: float = 40,
+    gamma: float = 3.80,
+    gamma_gain: float = 0.10,
+    log_gain: float = 1.00,
+    pmin: int = 80,
+    pmax: float = 0.3,
+    threshold: int = 30,
+) -> np.ndarray:
+    """Performs a series of denoiseing, filtering, and exposure adjustments to create a binary
+    mask for the given input image.
+
+    Args:
+        input_image (np.ndarray): The image to perform the streak masking on.
+        gaussian_sigma (float, optional): Parameter for `skimage.filters.gaussian`. Defaults to
+        40.
+        gamma (float, optional): Parameter for `skimage.exposure.adjust_gamma`. Defaults to 3.80.
+        gamma_gain (float, optional): Parameter for `skimage.exposure.adjust_gamma`. Defaults to
+        0.10.
+        log_gain (float, optional): Parameter for `skimage.exposure.adjust_log`. Defaults to 1.00.
+        pmin (int, optional): Lower bound for the `np.percentile` threshold, used for rescaling
+        the intensity. Defaults to 80.
+        pmax (int, optional): Upper bound for the `np.percentile` threshold, used for rescaling
+        the intensity. Defaults to 95.
+        threshold(float, optional): The lower bound for pixel values used to create a binary mask.
+        Defaults to 0.30.
+
+    Returns:
+        np.ndarray: The binary mask containing all of the candidate strokes.
+    """
+    # Denoise the Image
+    input_image = restoration.denoise_wavelet(
+        input_image, wavelet="db2", mode="hard", rescale_sigma=True
+    )
+    # Rescale the intensity using percentile ranges
+    pmin_v, pmax_v = np.percentile(input_image, (pmin, pmax))
+    input_image = exposure.rescale_intensity(input_image, in_range=(pmin_v, pmax_v))
+    # Laplace filter to get the streaks
```

Comment:
Add a space above the comments to help break up code a bit
```suggestion

    # Laplace filter to get the streaks
```